sway-features/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use clap::{Parser, ValueEnum};
+
+macro_rules! features {
+    (count; ) => {0};
+    (count; $name:ident $($rest:tt)?) => {1 + features!{count; $($rest)*}};
+    ($($name:ident = $enabled:literal, $url:literal),* $(,)?) => {
+        paste::paste! {
+            pub const CFG: [&str; features!{count; $($name)*}] = [
```

Comment:
We can have `pub const CFG: &[&str] = &["f1", "f2"];` here. In that case we don't need the recursive macro call just to calculate the size of the array. It simplifies the logic of the macro.

Reply:
That is actually a great idea! Done.